pkg/lisafs/handlers.go

```diff
@@ -1069,6 +1070,36 @@ func ConnectHandler(c *Connection, comm Communicator, payloadLen uint32) (uint32
 	return 0, nil
 }
 
+// ConnectWithCredsHandler handles the ConnectWithCreds RPC.
+func ConnectWithCredsHandler(c *Connection, comm Communicator, payloadLen uint32) (uint32, error) {
+	var req ConnectWithCredsReq
+	if _, ok := req.CheckedUnmarshal(comm.PayloadBuf(payloadLen)); !ok {
+		return 0, unix.EIO
+	}
+
+	fd, err := c.lookupControlFD(req.FD)
+	if err != nil {
+		return 0, err
+	}
+	defer fd.DecRef(nil)
+	if !fd.IsSocket() {
+		return 0, unix.ENOTSOCK
+	}
+	var sock int
+	if err := fd.safelyRead(func() error {
+		if fd.node.isDeleted() {
+			return unix.EINVAL
+		}
+		sock, err = fd.impl.ConnectWithCreds(req.SockType, req.UID, req.GID)
+		return err
+	}); err != nil {
+		return 0, err
+	}
+
+	comm.DonateFD(sock)
+	return 0, nil
+}
+
```

Comment:
I cannot switch uid/gid at handler level, because in this file the uid/gid in `fsgofer.Config{}` are inaccessible: gofer related files depend on `handlers.go`, so any code in `handlers.go` cannot depend back on gofer causing a cyclic dependency.

I think some slight code duplication with `ConnectHandler` here is acceptable.

Reply:
You should not switch uid/gid here. You need to switch it in this handler: https://github.com/google/gvisor/blob/0fb4cc71ba95bdc1e12ae87ff81476207a73b052/runsc/fsgofer/lisafs.go#L792-L793 (which is invoked by the lisfas/handlers.go file)

Reply:
Ah yes that's exactly what I did. Thanks for pointing this out